src/runner/general_settings.cpp

```diff
@@ -200,6 +205,13 @@ void apply_general_settings(const json::JsonObject& general_configs, bool save)
         settings_theme = general_configs.GetNamedString(L"theme");
     }
 
+    if (json::has(general_configs, L"hide_tray_icon", json::JsonValueType::Boolean))
+    {
+        hide_tray_icon = general_configs.GetNamedBoolean(L"hide_tray_icon");
+        // Update tray icon visibility when setting is toggled
+        set_tray_icon_visible(!hide_tray_icon);
+    }
+
```

Comment:
The addition of hide_tray_icon in apply_general_settings is well-implemented, and toggling the tray icon visibility based on this setting improves user control. Consider adding a brief comment to explain the specific use case or behavior, as it may help maintainers understand its purpose during future updates.